### PR TITLE
module village with location

### PR DESCRIPTION
**Pull request by ayocodingit**
# overview

- add village with location (module village)

image

Related: https://airtable.com/appdN4AdnU8FNs94M/tblRtWbyq60Fx1m6v/viwz1FKMjbiJpUkeB/rec6Qf99mDjjpiaaI?blocks=hide

cc: @jabardigitalservice/jds-backend 

Next Todo: 
- improve redis for maps locations

## Evidence
- title: add village with location (module village) peta sebaran desa digital 
- project: Desa Digital
- participants: @ayocodingit @firmanJS @sandisunandar99 @tukangremot @setiadijoe @rachadiannovansyah

### CODE REVIEW

File: src/modules/villages/village_repository.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import database from '../../config/database';
+import { Village as Entity } from './village_entity';
+
+export namespace Village {
+  export const Villages = () => database<Entity.Struct>('villages')
+
+  export const findAllWithLocation = () => {
```
Comment (setiadijoe) [inline src/modules/villages/village_repository.ts:7]: just ask, is there no filter at all from users?

**Review by setiadijoe — APPROVED** [review]

**Review by firmanJS — APPROVED** [review]

File: src/modules/villages/village_repository.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    const query = Villages()
+      .select('id', 'name', 'level', 'location')
+      .where('is_active', true)
+      .orderBy('name', 'asc')
+
+    return query
+  }
```
Comment (AldiNFitrah) [inline src/modules/villages/village_repository.ts:14]: Seems like we also need to return the category name
image

**Review by hafizulf — APPROVED** [review]

**Comment by tukangremot** [comment]
please follow this mock API : 

https://842cdb3d-b780-4a81-9c36-c9b6dce82ae2.mock.pstmn.io/v1/villages/list-with-location

File: src/modules/villages/village_entity.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+export namespace Village {
+  export interface Struct {
+    id: string
+    name: string
+    district_id: string
```
Comment (sandisunandar99) [inline src/modules/villages/village_entity.ts:5]: district_id is required or not ? if required you can add `?` in this field

**Review by tukangremot — APPROVED** [review]
mantuls..

**Pull request merged**